roachtest: acceptance/status-server failed

[(roachtest).acceptance/status-server failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1891147&tab=buildLog) on [provisional_202004210018_v20.1.0@c0f00d2a156295a1cb79ed960940c167cb03ccbc](https://github.com/cockroachdb/cockroach/commits/c0f00d2a156295a1cb79ed960940c167cb03ccbc):

```
The test failed on branch=provisional_202004210018_v20.1.0, cloud=gce:
test artifacts and logs in: artifacts/acceptance/status-server/run_1
	cluster.go:1864,status_server.go:28,acceptance.go:85,test_runner.go:753: error with attached stack trace:
		    main.execCmd
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:406
		    main.(*cluster).StartE
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:1859
		    main.(*cluster).Start
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:1864
		    main.runStatusServer
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/status_server.go:28
		    main.registerAcceptance.func1
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/acceptance.go:85
		    main.(*testRunner).runTest.func2
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s returned:
		    stderr:
		    %s
		    stdout:
		    %s
		    -- arg 1: <string>
		    -- arg 2: <string>
		    -- arg 3: <string>
		  - /go/src/github.com/cockroachdb/cockroach/bin/roachprod start local returned:
		    stderr:
		    ckroach/pkg/cmd/roachprod/install.(*SyncedCluster).Parallel.func1.1
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/install/cluster_synced.go:1622
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357: 
		    3: invalid version string 'c0f00d2'
		    github.com/cockroachdb/cockroach/pkg/util/version.Parse
		    	/go/src/github.com/cockroachdb/cockroach/pkg/util/version/version.go:90
		    github.com/cockroachdb/cockroach/pkg/cmd/roachprod/install.getCockroachVersion
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/install/cockroach.go:96
		    github.com/cockroachdb/cockroach/pkg/cmd/roachprod/install.Cockroach.Start.func2
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/install/cockroach.go:168
		    github.com/cockroachdb/cockroach/pkg/cmd/roachprod/install.(*SyncedCluster).Parallel.func1.1
		    	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/install/cluster_synced.go:1622
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357: 
		    I200422 03:21:14.225510 1 cluster_synced.go:1704  command failed
		    
		    stdout:
		    local: starting:
		  - exit status 1

```

<details><summary>More</summary><p>


Artifacts: [/acceptance/status-server](https://teamcity.cockroachdb.com/viewLog.html?buildId=1891147&tab=artifacts#/acceptance/status-server)
Related:
- #47809 roachtest: acceptance/status-server failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-master](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-master) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aacceptance%2Fstatus-server.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/alter_table.go
+++ b/pkg/sql/alter_table.go
@@ -825,7 +825,7 @@ func applyColumnMutation(
 			if err != nil {
 				return err
 			}
-			s := tree.Serialize(t.Default)
+			s := tree.Serialize(expr)
 			col.DefaultExpr = &s
 
 			// Add references to the sequence descriptors this column is now using.

--- a/pkg/sql/logictest/testdata/logic_test/pg_catalog
+++ b/pkg/sql/logictest/testdata/logic_test/pg_catalog
@@ -2564,3 +2564,24 @@ UPDATE pg_catalog.pg_tables SET a = 'abc'
 statement error pg_tables is a system catalog
 TRUNCATE TABLE pg_catalog.pg_tables
 
+# Regression for #47285.
+statement ok
+CREATE TABLE t47285 (x STRING DEFAULT 'hello');
+ALTER TABLE t47285 ALTER COLUMN x SET DEFAULT 'howdy'
+
+query T
+SELECT
+  pg_get_expr(d.adbin, d.adrelid)
+FROM
+  pg_attribute AS a
+  LEFT JOIN pg_attrdef AS d ON a.attrelid = d.adrelid AND a.attnum = d.adnum
+  LEFT JOIN pg_type AS t ON a.atttypid = t.oid
+  LEFT JOIN pg_collation AS c ON
+    a.attcollation = c.oid AND a.attcollation != t.typcollation
+WHERE
+  a.attrelid = 't47285'::REGCLASS
+  AND a.attnum > 0
+  AND NOT a.attisdropped
+  AND attname = 'x'
+----
+'howdy'::STRING

